src/state/ducks/posts/selectors.js

```diff
@@ -0,0 +1,37 @@
+const sort = (posts, sortBy, source) => {
+  if (source && posts[source] !== undefined) {
+    return sortBy === 'new' ?
+      posts[source].sort(post => new Date(post.timestamp) - new Date()) :
+      posts[source].sort((postF, postS) =>
+        (postS.meta.numViewed + postS.meta.numShared + postS.meta.numSaved) -
+        (postF.meta.numViewed + postF.meta.numShared + postF.meta.numSaved));
+  }
+  return sortBy === 'new' ?
+    posts.sort(post => new Date(post.timestamp) - new Date()) :
+    posts.sort((postF, postS) =>
+      (postS.meta.numViewed + postS.meta.numShared + postS.meta.numSaved) -
+      (postF.meta.numViewed + postF.meta.numShared + postF.meta.numSaved));
+};
+
+const getHomePosts = state => sortBy => (
+  sort(Object.values(state.posts.home), sortBy)
+);
+
+const getForYouPosts = state => sortBy => (
+  sort(Object.values(state.posts.forYou), sortBy)
+);
+
+const getSavedPosts = state => sortBy => (
+  sort(Object.values(state.posts.saved), sortBy)
+);
+
+const getSourcePosts = state => (sortBy, source) => (
+  state.posts.sources[source] ? sort(state.posts.sources, sortBy, source) : 0
```

Comment:
should return an empty array instead of `0`

Reply:
and why don't we just pass `state.posts.sources[source]` instead of `sort(state.posts.sources, ...` !?

the solution should be `sort(state.posts.sources[source] || [], sortBy)`

and refactoring the `sort` function is needed for this change. The sort function is just for sorting an inputted posts array, so only 2 params `(posts, sortBy)` are required.